editor/editor_vcs_interface.cpp

```diff
@@ -418,8 +418,13 @@ void EditorVCSInterface::create_vcs_metadata_files(VCSMetadata p_vcs_metadata_ty
 		if (f.is_null()) {
 			ERR_FAIL_MSG("Couldn't create .gitattributes in project path.");
 		} else {
-			f->store_line("# Normalize EOL for all files that Git considers text files.");
+			f->store_line("# Properly detect languages on GitHub");
+			f->store_line("*.gd linguist-language=GDScript");
```

Comment:
This is not necessary, GitHub will already detect GDScript without issues. We don't have or want manual defines for other languages that are already auto-detected like `*.cs linguist-language=C#`.

Reply:
> This is not necessary, GitHub will already detect GDScript without issues. We don't have or want manual defines for other languages that are already auto-detected like `*.cs linguist-language=C#`.

https://github.com/search?l=&o=desc&q=Godot+language%3AGAP&s=updated&type=Repositories

It's really hit or miss, I had to add this to my own project for it to work. I guess no language clashes with .cs that's why it works?

Reply:
That's odd, have you reported this as a bug on the Linguist repo? https://github.com/github/linguist

Reply:
Nope, but I just checked and saw your comment on an issue there about making it work with gdscript 2.

I haven't actually tested with 2 I just assumed it would be the same, it's hard to actually tell whether it works or not because it doesn't turn into GAP every time.

But if it's fixed then we could probably close this PR.

Reply:
It looks like it's broken on Godot's side, and there is already an issue open: https://github.com/godotengine/godot-vscode-plugin/issues/415

Reply:
That's good news, i'll close this pr. This was the main reason for it because it would save me from having to manually add linguist-lang every project.

Since it's actually an error they should just fix the problem.